llm/generate/gen_linux.sh

```diff
@@ -79,6 +79,7 @@ if [ -d /usr/local/cuda/lib64/ ]; then
         ${CUDA_LIB_DIR}/libcublasLt_static.a \
         ${CUDA_LIB_DIR}/libcudadevrt.a \
         ${CUDA_LIB_DIR}/libculibos.a \
+        -lcuda \
```

Comment:
This does beg the question of if we should just switch all these .a's to -l dep's and reduce our download size down by a few hundred MB.

Reply:
Dug a bit deeper into the package dependencies, and I think we should decouple this library vs. the other cuda libraries.   On ubuntu as a reference, if you just install the `cuda-driver` meta-package, that ties to a specific driver version (cuda-drivers-545 at present) and that package has a dependency on `libnvidia-compute-545` which delivers this libcuda.so.1 library.  So libraries like libcudart and friends are optional and many users wont have them installed, but if you have the nvidia driver, you should have libcuda.